### PR TITLE
Add versioning of blocks

### PR DESCRIPTION
**Pull request by i1i1**
Signed-off-by: i1i1 <vanyarybin1@live.ru>

### CODE REVIEW

**Review by eadventurous — COMMENTED** [review]
We also need to use `VersionedValidBlock` in sumeragi consensus. So that peers can be updated without the whole network being stopped.

**Comment by eadventurous** [comment]
Do you want to add these functionality in this or in a separate PR?

**Comment by i1i1** [comment]
Yeah, almost forgot about it. Will be in this pr soon!

File: iroha/src/kura.rs [hunk 1]
```diff
@@ -155,31 +182,35 @@ impl BlockStore {
...
         let metadata = metadata(&path).await.wrap_err("Unable to read metadata.")?;
         let mut buffer = vec![0; metadata.len() as usize];
         let _ = file.read(&mut buffer).await.wrap_err("Buffer overflow.")?;
-        Ok(ValidBlock::try_from(buffer).expect("Failed to read block from store."))
+        VersionedValidBlock::decode_versioned(&buffer).wrap_err("Failed to read block from store.")
     }
 
     /// Returns a sorted vector of blocks starting from 0 height to the top block.
-    async fn read_all(&self) -> Vec<ValidBlock> {
+    async fn read_all(&self) -> Result<Vec<ValidBlock>> {
```
Comment (humb1t) [inline iroha/src/kura.rs:203]: Seems like `read` and `read_all` should both return one of these block's types: `VersionedValidBlock` or `ValidBlock`

**Review by humb1t — CHANGES_REQUESTED** [review]

**Review by takemiyamakoto — APPROVED** [review]

**Review by humb1t — APPROVED** [review]

**Review by takemiyamakoto — APPROVED** [review]

File: iroha/src/block_sync.rs [hunk 2]
```diff
@@ -179,8 +179,15 @@ pub mod message {
                     }
                 }
                 Message::GetBlocksAfter(hash, peer) => {
-                    if block_sync.batch_size > 0 {
-                        if let Some(blocks) = block_sync.kura.read().await.blocks_after(*hash) {
+                    if block_sync.batch_size == 0 {
+                        log::error!(
+                            "Error: not sending any blocks as batch_size is equal to zero."
+                        );
```
Comment (eadventurous) [inline iroha/src/block_sync.rs:185]: Maybe it should be `warn`?

File: iroha/src/block_sync.rs [hunk 3]
```diff
@@ -195,9 +202,10 @@ pub mod message {
                                 }
                             }
                         }
-                    } else {
-                        log::error!("Error: not sending any blocks as batch_size is equal to zero.")
-                    }
+                        None => log::error!(
+                            "Error: not sending any blocks as batch_size is equal to zero."
+                        ),
```
Comment (eadventurous) [inline iroha/src/block_sync.rs:207]: Also should be `warn` to my mind, also please change the error message. Here the message should be that there are no blocks after `hash`.

File: iroha/src/queue.rs [hunk 4]
```diff
@@ -36,12 +37,18 @@ impl Queue {
     }
 
     /// Puts new transaction into queue. Returns error if queue is full.
-    pub fn push_pending_transaction(&mut self, tx: AcceptedTransaction) -> Result<()> {
+    pub fn push_pending_transaction(&mut self, tx: VersionedAcceptedTransaction) -> Result<()> {
```
Comment (eadventurous) [inline iroha/src/queue.rs:40]: Please move queue simplification into a separate PR.

File: iroha_version/iroha_version_derive/tests/json.rs [hunk 5]
```diff
@@ -73,21 +72,8 @@ mod tests {
 
         use model_1::*;
 
-        let raw_string = "{\"version\":\"3\",\"content\":\"test string\"}";
```
Comment (eadventurous) [inline iroha_version/iroha_version_derive/tests/json.rs:76]: It is important to compare raw content for the test.

Comment (eadventurous) [reply iroha_version/iroha_version_derive/tests/json.rs:76]: This should not be removed.

File: iroha_version/iroha_version_derive/tests/scale.rs [hunk 6]
```diff
@@ -72,21 +71,8 @@ mod tests {
         };
 
         use model_1::*;
-        let raw_string = "test string".encode();
```
Comment (eadventurous) [inline iroha_version/iroha_version_derive/tests/scale.rs:75]: Also here

**Review by eadventurous — CHANGES_REQUESTED** [review]

**Review by eadventurous — APPROVED** [review]

**Pull request merged**